lms/services/organization.py

```diff
@@ -136,28 +136,36 @@ def auto_assign_organization(
 
         if application_instance.organization:
             org = application_instance.organization
-
-        elif (
-            orgs := self._organization_search_query(guid=guid)
-            .order_by(Organization.updated.desc())
-            .all()
-        ):
-            if len(orgs) > 1:
-                LOG.warning(
-                    "Multiple organization matches found for application instance %s",
-                    application_instance.id,
-                )
-
-            org = orgs[0]
-
         else:
-            org = self.create_organization()
+            # Add a note to indicate the application instance was automatically
+            # allocated to an organization
+            application_instance.settings.set("flag", "auto_assigned_to_org", True)
```

Comment:
Not super pretty, but it's easy and doesn't involve a migration. Open to suggestions for a better name if you have one?

Reply:
We have notes in "hypothesis" -> "notes"

Not sure if `hypothesis" -> "auto_assigned_to_org" is better :shrug:

Reply:
Yep, if we have a place already we might as well re-use it